### PR TITLE
KEP: Promote server-side get APIs to GA

### PR DESCRIPTION
**Pull request by smarterclayton**
Initial draft of KEP.

### CODE REVIEW

**Comment by smarterclayton** [comment]
/assign @liggitt 
@kubernetes/sig-api-machinery-api-reviews @kubernetes/sig-cli-api-reviews

**Comment by smarterclayton** [comment]
GC controller needs support of PUT PartialObjectMetadata to fully switch, code change isn’t that bad.  Namespace controller needs nothing.  Quota needs nothing.  Serialization in proto of partialobjectmetadatalist is wrong, that might be worthy of a break (but indicates no one is likely using it because you can’t watch from it).

**Comment by wojtek-t** [comment]
> but indicates no one is likely using it because you can’t watch from it

@smarterclayton - why watch doesn't work now? I think ability to watch is quite important, so we should probably work on making that happen.

File: keps/sig-api-machinery/20190322-server-side-get-to-ga.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+### 1.16
+
+* Announce deprecation of `v1beta1` objects and removal in 1.17. 
+* `kubectl` should switch to using `meta.k8s.io/v1` `Table` (supporting 1.15 and 1.16 clusters)
+* Update at least a second controller to use `PartialObjectMetadata` `v1` and move the previous
+
+### 1.17
+
+* Remove `meta.k8s.io/v1beta1`
```
Comment (lavalamp) [inline keps/sig-api-machinery/20190322-server-side-get-to-ga.md]: Assuming it hasn't grown anything new?

File: keps/sig-api-machinery/20190322-server-side-get-to-ga.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+### Implementation Details
+
+A new dynamic client variant capable of supporting read and write operations on PartialObjectMetadata
+should be created that hides whether the server supports PartialObjectMetadata. 
+
+Currently `v1beta1.Table` does not support Protobuf and the generators do not trivially support the
+serialization of the cells. We need to decide on a serialization format for the Protobuf cells and
+ensure generators can be made to support it. This work does not need to block `v1` for Protobuf
+because the clients that access `Tables` are almost exclusively JSON clients (web consoles and CLIs).
```
Comment (lavalamp) [inline keps/sig-api-machinery/20190322-server-side-get-to-ga.md]: Seems potentially important for controllers just wanting partialObjectMetadata, though?

**Comment by lavalamp** [comment]
Sorry, I did not realize my lgtm would cause an insta-merge--I intended to leave this open for more discussion!

That said, I don't think anything here should be controversial, so I'm not sorry, either.

File: keps/sig-api-machinery/20190322-server-side-get-to-ga.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+Server-side columnar formatting and partial object metadata has been in beta since Kube 1.10 and as of 1.15 is consistently implemented and in wide use as part of `kubectl` and other web interfaces. This document outline required steps to graduate it to GA. 
+
+## Motivation
+
+The user of server-side printing with CRDs and Aggregated APIs is common and is a key part of providing equivalence between built-in and extension APIs for administrator usability. We have not needed to update the schema since 1.10, proving that it is ready to ship. Promoting it to GA in 1.15 will allow us to declare the kubectl portion feature complete and remove the legacy printers, and also update controllers that would benefit from use of PartialObjectMetadata (garbage collector, namespace controller, and quota counter) without fear of deprecation. PartialObjectMetadata allows these controllers to perform protobuf list actions efficiently when retrieving only the core object metadata.
+
+### Goals
+
+Server-side printing has no outstanding feature requests now that full WATCH support has been implemented in 1.15. It is ready to move to GA by promoting the resources.
```
Comment (wojtek-t) [inline keps/sig-api-machinery/20190322-server-side-get-to-ga.md]: @smarterclayton - you're talking about this one: https://github.com/kubernetes/kubernetes/pull/71548, right?

File: keps/sig-api-machinery/20190322-server-side-get-to-ga.md (hunk 2 above)
Comment (smarterclayton) [reply keps/sig-api-machinery/20190322-server-side-get-to-ga.md]: Controllers absolutely need proto and that works "ok"

File: keps/sig-api-machinery/20190322-server-side-get-to-ga.md (hunk 3 above)
Comment (smarterclayton) [reply keps/sig-api-machinery/20190322-server-side-get-to-ga.md]: Yes, thank you will link.

**Comment by smarterclayton** [comment]
It's in initial state

**Comment by smarterclayton** [comment]
Implementable update in #929

**Pull request merged**